plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php

```diff
@@ -37,22 +38,23 @@ public function __construct() {
 	public static function get_suggestions( array $specs = null ) {
 		$locale = get_user_locale();
 
-		$specs   = is_array( $specs ) ? $specs : self::get_specs();
-		$results = EvaluateSuggestion::evaluate_specs( $specs );
+		$specs           = is_array( $specs ) ? $specs : self::get_specs();
+		$results         = EvaluateSuggestion::evaluate_specs( $specs );
+		$specs_to_return = $results['suggestions'];
 
 		// When suggestions is empty, replace it with defaults and save for 3 hours.
-		if ( empty( $results['suggestions'] ) ) {
+		if ( empty( $specs_to_return ) ) {
 			PaymentGatewaySuggestionsDataSourcePoller::get_instance()->set_specs_transient( array( $locale => DefaultPaymentGateways::get_all() ), 3 * HOUR_IN_SECONDS );
-
-			return EvaluateSuggestion::evaluate_specs( DefaultPaymentGateways::get_all() )['suggestions'];
+			$specs_to_return = EvaluateSuggestion::evaluate_specs( DefaultPaymentGateways::get_all() )['suggestions'];
```

Comment:
Since we don't return the specs here, when suggestions are empty and errors exist, we will set the transient to `$specs` in line 53. Is this an intentional change?

Reply:
Thanks for catching it! It's not intentional, the logic got over my head 😵

Reply:
@chihsuan I've updated the specs logic with https://github.com/woocommerce/woocommerce/pull/44775/commits/c39de1a7d5c7b950b8dacc6079587032725fbe7c